src/main/java/net/neoforged/neoforge/common/damagesource/DamageContainer.java

```diff
@@ -0,0 +1,304 @@
+
+    /**
+     * This sets the current damage value for the entity at the stage of the damage sequence in which it is set.
+     * Subsequent steps in the damage sequence will use and modify this value accordingly. If this is called in
+     * the final step of the sequence, this value will be applied against the entity's health.
+     *
+     * @param damage the amount to harm this entity at the end of the damage sequence
+     */
+    void setNewDamage(float damage);
+
+    /**
+     * @return The damage blocked during the {@link net.neoforged.neoforge.event.entity.living.DamageBlockEvent}
+     */
+    float getBlockedDamage();
+
+    /**
+     * @return The durability applied to the applicable shield after {@link net.neoforged.neoforge.event.entity.living.DamageBlockEvent}
+     *         returned a successful block
+     */
+    float getShieldDamage();
+
+    /**
+     * Explicitly sets the invulnerability ticks after the damage has been applied.
+     *
+     * @param ticks Ticks of invulnerability after this damage sequence
+     */
+    void setPostAttackInvulnerabilityTicks(int ticks);
+
+    /**
+     * @return The number of ticks this entity will be invulnerable after damage is applied.
+     */
+    int getPostAttackInvulnerabilityTicks();
+
+    /**
+     * This provides a post-reduction value for armor reduction and modifiers. This will always return zero
+     * before {@link net.neoforged.neoforge.event.entity.living.IncomingDamageEvent} and will consume all
+     * modifiers prior to the event.
+     *
+     * @return The amount of damage reduced by armor after vanilla armor reductions and added modifiers
+     */
+    float getArmorReduction();
+
+    /**
+     * This provides a post-reduction value for enchantment reduction and modifiers. This will always return zero
+     * before {@link net.neoforged.neoforge.event.entity.living.IncomingDamageEvent} and will consume all
+     * modifiers prior to the event.
+     *
+     * @return the amount of damage reduced by enchantments after vanilla enchantment reductions and added modifiers
+     */
+    float getEnchantReduction();
+
+    /**
+     * This provides a post-reduction value for mob effect reduction and modifiers. This will always return zero
+     * before {@link net.neoforged.neoforge.event.entity.living.IncomingDamageEvent} and will consume all
+     * modifiers prior to the event.
+     *
+     * @return The amount of damage reduced by mob effects after vanilla mob effect reductions and added modifiers
+     */
+    float getMobEffectReduction();
+
+    /**
+     * This provides a post-reduction value for absorption consumption and modifiers. This will always return zero
+     * before {@link net.neoforged.neoforge.event.entity.living.IncomingDamageEvent} and will consume all
+     * modifiers prior to the event.
+     *
+     * @return The amount of absorption consumed after vanilla absorption consumption and added modifiers
+     */
+    float getAbsorption();
```

Comment:
Given the `Reduction` enum, should these be collapsed into one method with a `Reduction` parameter?